.github/workflows/test-in-app-messaging-prs.yml

```diff
@@ -0,0 +1,63 @@
+# Description: This workflow runs unit + e2e tests on PRs targeting
+#   - `in-app-messaging/main`
+#   - `in-app-messaging/release`
+#
+# Triggered by:
+#   (1) Internal PRs: maintainer pushes a commit to PRs targeting those branches.
+#   (2) Fork PRs: maintainer adds "run-test" label to PRs targeting those branches.
+
+name: Test / in-app-messaging / PRs
+
+concurrency:
+  group: e2e-${{ github.event.pull_request.id }}
+  cancel-in-progress: true
+
+on:
+  pull_request_target:
+    branches: [
+        in-app-messaging/main
+        in-app-messaging/release,
+      ]
+    types: [opened, synchronize, labeled]
+
+permissions:
+  pull-requests: write # used to remove label
+  # other permissions are defaulted to "none"
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    # We run tests only if it's:
+    #   1) pull request not from a fork (ie. internal PR), or
+    #   2) pull request from a fork (ie. external PR) that was added "run-tests" label
+    if: |
+      (github.event.pull_request.head.repo.full_name == github.repository) ||
+      (github.event.action == 'labeled' && github.event.label.name == 'run-tests')
+    steps:
+      - name: Remove run-tests label, if applicable
+        if: always() && github.event.label.name == 'run-tests'
```

Comment:
for my own understanding, why is the `always()` call required here? Seems like only the latter conditional would be required.

Reply:
Ooh, that's a great question. 

This is a left over logic from when there was one monolithic testing workflow for PRs, `main`, etc. 

`always() && github.event.label.name == 'run-tests'` made sure this step always runs, even if the previous steps failed. I did that to feel 100% about security.

Now that we separated `test-prs` out with reusable workflows, this is no longer necessary. I'll create a separate PR that cleans it all up! No later than EOD.